### PR TITLE
Fix lint for SequentialReadingService and unify the API for process_reset_fn

### PR DESCRIPTION
**Pull request by ejguan**
This is the follow up for 807db8f8c7282b2f48b48b1e07439c119a2ba12f

### Changes

- Fix lint
- Fix checkpoint result
- Add inline doc
- Unify the API for process_reset_fn and dispatch_process_reset_fn

After this PR, I will land a test case and tutorial for SequentialRS

### CODE REVIEW

File: torchdata/dataloader2/reading_service.py [hunk 1]
```diff
@@ -303,7 +310,9 @@ def initialize(self, datapipe: DataPipe) -> DataPipe:
 
         return self._end_datapipe  # type: ignore[return-value]
 
-    def initialize_iteration(self, seed_generator: SeedGenerator, iter_reset_fn: Optional[Callable[[DataPipe], DataPipe]] = None) -> Optional[Callable[[DataPipe], DataPipe]]:
+    def initialize_iteration(
+        self, seed_generator: SeedGenerator, iter_reset_fn: Optional[Callable[[DataPipe], DataPipe]] = None
+    ) -> Optional[Callable[[DataPipe], DataPipe]]:
```
Comment (NivekT) [inline torchdata/dataloader2/reading_service.py:315]: Question: where is the case when it returns something other than `None`?

File: torchdata/dataloader2/reading_service.py [hunk 2]
```diff
@@ -540,9 +560,12 @@ def checkpoint(self) -> bytes:
         states = []
         for rs in self.reading_services:
             states.append(rs.checkpoint())
+        return b"\n".join(states)
```
Comment (NivekT) [inline torchdata/dataloader2/reading_service.py:563]: This is more of a placeholder for now? To be changed later?

**Review by NivekT — APPROVED** [review]
LGTM with some questions!

File: torchdata/dataloader2/reading_service.py (hunk 2 above)
Comment (ejguan) [reply torchdata/dataloader2/reading_service.py:563]: This is a valid solution to combine multiple `bytes`, unless you have any concern on this solution.

**Pull request closed without merging**